Service/FindOrganizationThroughRepositoriesService.php

```diff
@@ -8,42 +8,148 @@
 
+/**
+ * Loops through repositories (https://opencatalogi.nl/oc.repository.schema.json) and updates it with fetched organization info
+ */
 class FindOrganizationThroughRepositoriesService
 {
     private EntityManagerInterface $entityManager;
-    private GithubApiService $githubService;
     private array $configuration;
     private array $data;
+    private SymfonyStyle $io;
+    private CallService $callService;
+
+    private Entity $organisationEntity;
+    private Entity $repositoryEntity;
+    private Source $githubApi;
 
     public function __construct(
         EntityManagerInterface $entityManager,
-        GithubApiService $githubService
+        CallService $callService
     ) {
+        $this->callService = $callService;
         $this->entityManager = $entityManager;
-        $this->githubService = $githubService;
+
         $this->configuration = [];
         $this->data = [];
     }
 
     /**
-     * @param ObjectEntity $repository the repository where we want to find an organisation for
+     * Set symfony style in order to output to the console
      *
-     * @throws Exception
+     * @param SymfonyStyle $io
+     * @return self
      */
-    public function setRepositoryWithGithubInfo(ObjectEntity $repository, $github): ObjectEntity
+    public function setStyle(SymfonyStyle $io): self
     {
-        $repository->setValue('source', 'github');
-        $repository->setValue('name', $github['name']);
-        $repository->setValue('url', $github['url']);
-        $repository->setValue('avatar_url', $github['avatar_url']);
-        $repository->setValue('last_change', $github['last_change']);
-        $repository->setValue('stars', $github['stars']);
-        $repository->setValue('fork_count', $github['fork_count']);
-        $repository->setValue('issue_open_count', $github['issue_open_count']);
-        $repository->setValue('programming_languages', $github['programming_languages']);
+        $this->io = $io;
 
+        return $this;
+    }
+
+    
+
+    /**
+     * This function gets the content of the given url.
+     *
+     * @param string      $url
+     * @param string|null $path
+     *
+     * @return array|null
+     */
+    public function requestFromUrl(string $url, ?string $path = null): ?array
+    {
+        // @TODO callService needs to be used without source, full domain here is variable
+        // if ($path !== null) {
+        //     $parse = parse_url($url);
+        //     $url = str_replace([$path], '', $parse['path']);
+        // }
+
+        // if ($response = $this->callService->call('GET', $url)) {
+        //     return json_decode($response->getBody()->getContents(), true);
+        // }
+
+        return null;
+    }
+
+    /**
+     * This function gets all the github repository details.
+     *
+     * @param array $item a repository from github with a publicclode.yaml file
+     *
+     * @return array
+     */
+    public function getGithubRepositoryInfo(array $item): array
```

Comment:
Wat is het voordeel van dit doen versus het gebruiken van de mapping service of gewoon hydraten?